### PR TITLE
162633497 Validate user sign up errors

### PR DESCRIPTION
**Pull request by jeanjoe**
**What does this PR do?**

This PR contains Unit Tests for User Sign up validation Errors.

**Description of Tasks completed**

* Add missing dependencies

* Add name to user sign up end point

* Add custom validation for password strength

* Add tests for user sign up errors

**How to manually run the test.**
To run the tests, use this command when virtual environment is activated

`python manage.py test`

**Background context**
The following files where modified to make the tests run successfully
* `.travis.yml`
* `authours/urls.py`
* `authors/apps/authentication/urls.py`
* `authors/apps/authentication/serializers.py`

**Relevant pivotal tracker story**

[DELIVERS #162633497]
(https://www.pivotaltracker.com/n/projects/2231917/stories/162633497)

### CODE REVIEW

**Comment by mycok** [comment]
LGTM

**Comment by mubstimor** [comment]
LGTM

**Comment by Qadriyah** [comment]
LGTM

File: authors/apps/authentication/serializers.py [hunk 1]
```diff
@@ -3,7 +3,20 @@
...
-
+import re
+
+class CustomValidator:
+    def password_strength(value):
+        """
+        Test if password contains numbers and letters else raise serializer
+        Validation error
+        """
+        if (re.compile(r'^(?=.*[A-Za-z0-9])(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]').search(value['password'])) is None:
```
Comment (lym) [inline authors/apps/authentication/serializers.py]: This line is too long. Consider refactoring it to:
```suggestion
       if re.compile(
            r'^(?=.*[A-Za-z0-9])(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]'
        ).search(value['password']) is None:
```

File: authors/apps/authentication/serializers.py [hunk 2]
```diff
@@ -3,7 +3,20 @@
...
+            raise serializers.ValidationError(
+                {
+                    'password': "Password must contain a number, letters and a special character."
```
Comment (lym) [inline authors/apps/authentication/serializers.py]: This line is too long. Please fit it within 80 characters.

File: authors/apps/authentication/tests/test_user_sign_up_validation_errors.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        self.createUser(self.user)
+        response = self.createUser(usernameExistsData)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.data, usernameExistsResponse)
+    
+    def test_user_sign_up_password_length(self):
+        """
+        Test an if password length is less than 8 charaters.
+        """
+        passwordLenghtData = {
```
Comment (lym) [inline authors/apps/authentication/tests/test_user_sign_up_validation_errors.py]: Please stick to [snake case](https://en.wikipedia.org/wiki/Snake_case) for all your _variable_ and _method_ names.

**Comment by lym** [comment]
@jeanjoe Please address the comments and also squash all your commits into one.

File: authors/apps/authentication/tests/base.py [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
-                'email':'testuser@andela.com' , 
-                'password':'TestUser12#'
+            "user": {
+                "email":  "testuser@andela.com",
+                "username": 'testuser',
+                'password': 'TestUser12#',
             }
         }
 
+    def createUser(self, data):
```
Comment (lym) [inline authors/apps/authentication/tests/base.py]: Use snake case here.

File: authors/apps/authentication/tests/test_user_sign_up_validation_errors.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        self.createUser(self.user)
+        response = self.createUser(username_exists_data)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.data, username_exists_response)
+    
+    def test_user_sign_up_password_length(self):
+        """
+        Test an if password length is less than 8 charaters.
+        """
+        password_lenght_data = {
```
Comment (lym) [inline authors/apps/authentication/tests/test_user_sign_up_validation_errors.py]: Please correct the spelling of the word _length_ in the `password_lenght_data` variable name.

**Pull request merged**